pbs/pbsrequest.go

```diff
@@ -202,6 +231,28 @@ func ParsePBSRequest(r *http.Request, cache cache.Cache) (*PBSRequest, error) {
 			glog.Infof("Ad unit %s has %d bidders for %d sizes", unit.Code, len(bidders), len(unit.Sizes))
 		}
 
+		var mtypes []MediaType
```

Comment:
This looks like it could probably be a standalone function, `ParseMediaTypes(types []string) []MediaType`. Or maybe `ParseMediaTypes(types []string) ([]MediaType, error)`. I'm not sure whether Go conventions let you return both a usable value _and_ an error. All the code I've seen is either/or... but this looks like a case where both might be appropriate.

Either way: this function's already 128 lines long... which is at risk of being spaghetti code, if it hasn't crossed that line already. The code you're adding will be testable that way, too.